Doc/library/typing.rst

```diff
@@ -585,6 +587,22 @@ These can be used as types in annotations and do not support ``[]``.
    .. versionadded:: 3.5.4
    .. versionadded:: 3.6.2
 
+.. data:: Self
+
+   Special annotation to represent the current enclosed class.
```

Comment:
This seems like a bit weird wording to me. I think it is better to call `Self` a `type`, not `annotation`.

Reply:
This documentation is a bit sparse. I'd suggest adding a few points:
- It's useful for methods that return instances of the current class (common examples: classmethods serving as alternative constructors; `__enter__`)
- It's equivalent to a TypeVar with bound (but no need to write out the full equivalent form), just to point people to something similar if they want to understand it.